docs/extension-development/adding-custom-products-to-add-products-onboarding-list.md

```diff
@@ -0,0 +1,75 @@
+---
+post_title: How to add custom product types to Add Products onboarding list
+menu_title: Add custom product types to Add Products onboarding list
+tags: how-to
+---
+
+## Introduction
+
+WooCommerce allows developers to extend the product type onboarding list, offering a more customizable and engaging experience during the Add Products onboarding task. This tutorial will guide you through adding custom product types to your WooCommerce store using the `experimental_woocommerce_tasklist_product_types` JavaScript filter.
+
+## Prerequisites
+
+- A basic understanding of JavaScript and PHP.
+- WooCommerce 8.8 or later installed on your WordPress site.
+
+## Step 1: Adding a JavaScript Filter
+
+To add a new product type to the onboarding list, we'll utilize the `@wordpress/hooks` package, specifically the addFilter function. If you're not already familiar, `@wordpress/hooks` allows you to modify or extend features within the WordPress and WooCommerce ecosystem without altering the core code.
+
+First, ensure you have the `@wordpress/hooks` package installed. If not, you can add it to your project using `npm` or `yarn`:
+
+`npm install @wordpress/hooks`
+
+or:
+
+`yarn add @wordpress/hooks`
+
+Next, add the following JavaScript code to your project. This code snippet demonstrates how to add a "custom product" type to the onboarding list:
```

Comment:
It's helpful to mention where in the file-tree this code chunk or file needs to be added.

Reply:
@dakota one more, can you help here?

Reply:
@dakota i lost this one for a minute, are you able to help with this Q?

Reply:
There isn't really any specific place, it entirely depends on the file/project structure of the extension (or store) that will be using this.